ibis/pandas/client.py

```diff
@@ -321,6 +323,64 @@ def compile(self, expr, *args, **kwargs):
     def database(self, name=None):
         return PandasDatabase(name, self)
 
+    def get_schema(self, table_name, database=None):
+        """
+        Return a Schema object for the indicated table and database
+
+        Parameters
+        ----------
+        table_name : string
+          May be fully qualified
+        database : string, default None
+
+        Returns
+        -------
+        schema : ibis Schema
+        """
+        return infer_pandas_schema(self.dictionary[table_name])
+
+    def list_tables(self, like=None):
+        tables = list(self.dictionary.keys())
+        if like is not None:
+            return list(filter(lambda t: re.search(like, t), tables))
+        return tables
+
+    def load_data(self, table_name, obj, **kwargs):
+        """
+        Parameters
+        ----------
+        table_name : string
+        obj: pandas.DataFrame
+        """
+        # kwargs is a catch all for any options required by other backends.
+        self.dictionary[table_name] = pd.DataFrame(obj)
+
+    def create_table(self, table_name, obj=None, schema=None):
+        if obj is None and schema is None:
+            raise com.IbisError('Must pass expr or schema')
+
+        if obj is not None:
+            df = pd.DataFrame(obj)
+        else:
+            df = schema.apply_to(pd.DataFrame())
+
+        self.dictionary[table_name] = df
+
+    def exists_table(self, name):
+        """
+        Determine if the indicated table or view exists
+
+        Parameters
+        ----------
+        name : string
+        database : string, default None
+
+        Returns
+        -------
+        if_exists : boolean
+        """
+        return name in self.list_tables()
```

Comment:
I still think this should be `name in self.dictionary`. No reason to make this unnecessarily slow.